src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs

```diff
@@ -144,9 +144,7 @@ private static async Task<(Stream originalBody, Stream copy)> CopyBody(HttpReque
         {
             if (!location.IsAbsoluteUri && response.RequestMessage.RequestUri is Uri requestUri)
             {
-                location = new Uri(
-                    new Uri(requestUri.GetLeftPart(UriPartial.Authority)),
-                    location);
+                location = new Uri(requestUri, location);
```

Comment:
That change looks correct for me and will be consistent with what we have in the "real" `HttpClient`

https://github.com/dotnet/runtime/blob/56e821243d92f20a6abf7ca61288221f4d164dd1/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs#L112

My only concern if that will impact tests and we should add a **Breaking change announcement**.